official/resnet/resnet_run_loop.py

```diff
@@ -211,9 +214,17 @@ def resnet_model_fn(features, labels, mode, model_class,
   # Generate a summary node for the images
   tf.summary.image('images', features, max_outputs=6)
 
-  model = model_class(resnet_size, data_format, version=version)
+  if dtype == tf.float16:
+    features = tf.cast(features, tf.float16)
```

Comment:
As above: I think we can get away with just doing the cast to dtype? Check speed and check with @reedwm , but my guess is a no-op cast is fine, and saves us from conditionals. If that's the case, make sure to add a comment about why we are casting to avoid someone in the future thinking, gee, this is a no-op cast, because the features and dtype are both fp32.

Reply:
Discussed offline.